src/rust/src/buf.rs

```diff
@@ -19,6 +21,13 @@ fn _extract_buffer_length<'p>(
     mutable: bool,
 ) -> pyo3::PyResult<(pyo3::Bound<'p, pyo3::PyAny>, usize)> {
     let py = pyobj.py();
+    if pyobj.is_instance_of::<pyo3::types::PyString>() {
+        return Err(PyErr::new::<PyTypeError, _>(
+            "\n\
+             Cannot create a buffer from a string.\n\
+             Did you mean to pass a bytestring instead?",
+        ));
+    }
```

Comment:
Hmm, is there a reason to special case strings, as opposed to just raising our own error  anytime `FFI_FROM_BUFFER` fails?

Reply:
Good call, replaced this with a `map_err` on the `Result` returned by calling `get_buffer`.